controllers/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -112,197 +83,21 @@ func (r *Reconciler) ensureStatusUpdateRecord(logger logr.Logger, tier *toolchai
 	}
 	// if there was no previous status:
 	if len(tier.Status.Updates) == 0 {
-		tier.Status.Updates = append(tier.Status.Updates, toolchainv1alpha1.NSTemplateTierHistory{
-			StartTime: metav1.Now(),
-			Hash:      hash,
-		})
-		return true, r.Client.Status().Update(context.TODO(), tier)
+		return true, r.addNewTierUpdate(tier, hash)
 	}
-	// check that last entry
+	// check whether the entry was already added
 	if tier.Status.Updates[len(tier.Status.Updates)-1].Hash == hash {
-		logger.Info("current update (or check if update is needed) is still in progress")
+		logger.Info("current tier template already exists in tier.status.updates")
 		return false, nil
 	}
-	// reset the `FailedAccounts` in the previous update as we don't want to retain the usernames
-	// for whom the update failed previously (no need to carry such data anymore)
-	tier.Status.Updates[len(tier.Status.Updates)-1].FailedAccounts = nil
 	logger.Info("Adding a new entry in tier.status.updates")
+	return true, r.addNewTierUpdate(tier, hash)
+}
+
+func (r *Reconciler) addNewTierUpdate(tier *toolchainv1alpha1.NSTemplateTier, hash string) error {
 	tier.Status.Updates = append(tier.Status.Updates, toolchainv1alpha1.NSTemplateTierHistory{
 		StartTime: metav1.Now(),
```

Comment:
long-term, it would be fine to rename for example to "UpdateTime", but it's fine as it is for now